### PR TITLE
first examples of nested, parameterized testsets

### PR DESCRIPTION
**Pull request by sbromberger**
Here's what a failure looks like:
```
Test Summary:               | Pass  Fail  Error  Total
LightGraphs                 | 5727     2      1   5730
  LightGraphs.SimpleGraphs  | 4582     2          4584
    SimpleGraphs            |  607                 607
    SimpleEdge              |   33                  33
    SimpleEdgeIter          | 2430     2          2432
      string representation |    2                   2
      collection operations |   12                  12
      eltype                |    4                   4
      membership operations | 2401     2          2403
      iterator protocol     |    6                   6
      graph modifications   |    4                   4
    Randgraphs              | 1286                1286
    Static graphs           |  187                 187
    Small graphs            |   26                  26
    Euclidean graphs        |   13                  13
  LightGraphs.LinAlg        |  391                 391
  LightGraphs.Parallel      |  733                 733
  Interface                 |   20                  20
```

ref #1184

### CODE REVIEW

**Comment by sbromberger** [comment]
cc @simonschoelly

**Comment by jpfairbanks** [comment]
I think this is a good idea

**Comment by sbromberger** [comment]
@jpfairbanks @simonschoelly My inclination is to keep going on this via several smaller PRs (to avoid a massive PR at the end), but I want to merge each of them before I start on the next batch. 

Do you have any strong objections to this plan? I will not seek review for each PR.

File: test/runtests.jl [hunk 1]
```diff
@@ -11,8 +11,9 @@ using Statistics: mean
 
 const testdir = dirname(@__FILE__)
 
-testgraphs(g) = [g, Graph{UInt8}(g), Graph{Int16}(g)]
-testdigraphs(g) = [g, DiGraph{UInt8}(g), DiGraph{Int16}(g)]
+testgraphs(g) = is_directed(g) ? [g, DiGraph{UInt8}(g), DiGraph{Int16}(g)] : [g, Graph{UInt8}(g), Graph{Int16}(g)]
```
Comment (simonschoelly) [inline test/runtests.jl]: Maybe it would be easier, if we also made a copy of `g` so that we don't have to make this copy in every loop.

**Comment by simonschoelly** [comment]
I like the general direction of this. One thing, that also could do in cases where it is not exactly clear, what we are testing, is that we put what should be the result into the name of the testset.

See for example this article: https://dzone.com/articles/7-popular-unit-test-naming

It is for Java, and luckily we don't have to write such a thing in a method name, but can use full strings.

File: test/runtests.jl (hunk 1 above)
Comment (sbromberger) [reply test/runtests.jl]: not sure I understand. We return `g` in the vector....

**Comment by sbromberger** [comment]
I plan on merging this tomorrow or Friday unless there are strong objections or obvious mistakes :)

File: test/runtests.jl (hunk 1 above)
Comment (simonschoelly) [reply test/runtests.jl]: In the code we have something like
```
 @testset "$g" for g in testgraphs(gx, gdx) 
        @test @inferred(nv(g)) == 15
            gc = copy(g)
            @test add_vertices!(gc, 5) == 5
            @test @inferred(nv(gc)) == 15
        end
```
If we changed that line to
`testgraphs(g) = is_directed(g) ? [copy(g), DiGraph{UInt8}(g), DiGraph{Int16}(g)] : [copy(g), Graph{UInt8}(g), Graph{Int16}(g)]`
then we could omit the line `gc = copy(g)`.

**Comment by matbesancon** [comment]
this PR is a bit small, but let's say it's fine :trollface:

File: test/runtests.jl (hunk 1 above)
Comment (sbromberger) [reply test/runtests.jl]: I don't think that's right. We use `copy(g)` when we want to modify `g` (to do some checking, for example) within the loop created with`testgraphs`. If we just added a `copy(g)` to `testgraphs`, it wouldn't be a copy of the graph with the changed eltype.

**Pull request merged**